File: client2/src/game/components/movement.ts
import { Flag } from './flags'

type Direction = 'left' | 'right'

export class Movement {
  static _id = 'Movement'
  _id = Movement._id

  static _flag = Flag.Movement
  _flag = Flag.Movement

  direction: Direction = 'right'
  left: boolean = false
  right: boolean = false
  jump: boolean = false
  jumpCount: number = 0

  constructor() {}
}

File: archive/client/src/game/components/movement.lua
local constants = require 'src.game.components.constants'

local Movement = {
  _meta = constants.Movement
}

function Movement.new(id)
  local movement = {
    _meta = Movement._meta,
    id = id,
    direction = 'right', -- left or right
    left = false,
    right = false,
    jump = false,
    jumpCount = 0
  }

  return movement
end

return Movement
